airflow/settings.py

```diff
@@ -151,11 +151,29 @@ def configure_orm(disable_connection_pool=False):
     pool_connections = conf.getboolean('core', 'SQL_ALCHEMY_POOL_ENABLED')
     if disable_connection_pool or not pool_connections:
         engine_args['poolclass'] = NullPool
+        log.info("settings.configure_orm(): Using NullPool")
     elif 'sqlite' not in SQL_ALCHEMY_CONN:
-        # Engine args not supported by sqlite
-        engine_args['pool_size'] = conf.getint('core', 'SQL_ALCHEMY_POOL_SIZE')
-        engine_args['pool_recycle'] = conf.getint('core',
-                                                  'SQL_ALCHEMY_POOL_RECYCLE')
+        # Engine args not supported by sqlite.
+        # If no config value is defined for the pool size, select a reasonable value.
+        # 0 means no limit, which could lead to exceeding the Database connection limit.
+        try:
+            pool_size = conf.getint('core', 'SQL_ALCHEMY_POOL_SIZE')
+        except conf.AirflowConfigException:
```

Comment:
When is this exception thrown?

Reply:
Without more context, swallowing this exception seems like a bad idea.

Reply:
+1. If the config is not defined, let burn.

Reply:
It throws an exception if the config is not defined? In that case, never mind.